vars/deploy.groovy

```diff
@@ -0,0 +1,159 @@
+import groovy.transform.Field
+
+import static com.sap.piper.Prerequisites.checkScript
+
+@Field String STEP_NAME = getClass().getName()
+
+@Field Set GENERAL_CONFIG_KEYS = [
+     /** Defines the targets to deploy on cloudFoundry or neo.*/
+    'cfTargets',
+    'neoTargets'
+]
+
+@Field Set STEP_CONFIG_KEYS = []
+
+@Field Set PARAMETER_KEYS = GENERAL_CONFIG_KEYS.plus([
+     'stage',
+     /** Defines the deployment type */
+     'enableZeroDowntimeDeployment',
+    /** only for neo deployment*/
+    'source'
+])
+
+
+void call(parameters = [:]) {
+
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+
+        def index = 1
+        def deployments = [:]
+        def stageName = env.STAGE_NAME
+        def enableZeroDowntimeDeployment = parameters.enableZeroDowntimeDeployment
+
+        def script = checkScript(this, parameters) ?: this
+
+        def utils = parameters.utils ?: new Utils()
+        def jenkinsUtils = parameters.jenkinsUtils ?: new JenkinsUtils()
+ 
+        Map config = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .use()
+
+        utils.pushToSWA([
+            step: STEP_NAME,
+            stepParamKey1: 'cfTargets',
+            stepParam1: config.cfTargets,
+            stepParamKey2: 'neoTargets',
+            stepParam2: config.neoTargets
+        ], config)
+
+        if (config.cfTargets) {
+            for (int i = 0; i < config.cfTargets.size(); i++) {
+                def target = config.cfTargets[i]
+                Closure deployment = {
+                    utils.unstash(stageName)
+
+                    String deploymentType
+                    if(enableZeroDowntimeDeployment) {
+                        deploymentType = DeploymentType.CF_BLUE_GREEN.toString()
+                    }
+                    else {
+                        deploymentType = DeploymentType.selectFor(
+                            CloudPlatform.CLOUD_FOUNDRY,
+                            config.isProduction.asBoolean()
+                        ).toString()
+                    }
+
+                    def deployTool =
+                        (script.commonPipelineEnvironment.configuration.isMta) ? 'mtaDeployPlugin' : 'cf_native'
+
+                    cloudFoundryDeploy(
+                        script: script,
+                        juStabUtils: utils,
+                        jenkinsUtilsStub: jenkinsUtils,
+                        deployType: deploymentType,
+                        cloudFoundry: target,
+                        mtaPath: script.commonPipelineEnvironment.mtarFilePath,
+                        deployTool: deployTool
+                    )
+
+                    utils.stash(stageName)
+                }
+                deployments["Deployment ${index > 1 ? index : ''}"] = {
+                    if (env.POD_NAME) {

```

Comment:
Here we try to find out if we run on kubernetes, right? For that we have an envionmentVariable (`ON_K8S`). Would be more explict to use this.